### PR TITLE
`Ltac ::=`: respect `Local`

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by Alizter** [comment]
Test?

**Comment by SkySkimmer** [comment]
Should be ready

File: doc/changelog/05-tactic-language/16106-local-ltac-redef.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Changed:**
+  :cmd:`Ltac` redefinitions (with ``::=`` now respect :attr:`local`
```
Comment (Alizter) [inline doc/changelog/05-tactic-language/16106-local-ltac-redef.rst:2]:
```suggestion
  :cmd:`Ltac` redefinitions (with ``::=``) now respect :attr:`local`
```

File: doc/sphinx/proof-engine/ltac.rst [hunk 2]
```diff
@@ -1839,8 +1839,8 @@ Defining |Ltac| symbols
 
    Defines or redefines an |Ltac| symbol.
 
-   If the :attr:`local` attribute is specified, the definition will not be
-   exported outside the current module.
+   If the :attr:`local` attribute is specified, definitions will not be
+   exported outside the current module and redefinitions only apply for the current module.
```
[the remark this answers is not in the document]
Comment (SkySkimmer) [reply doc/sphinx/proof-engine/ltac.rst:1843]: Depends what exported means. Let's keep the current wording so we don't have to care.

**Comment by Alizter** [comment]
When you do a redefinition without `Local` in a module, what happens when you require and when you import?

**Comment by ppedrot** [comment]
Do we have a test for section-local redefinitions? If not it might be the right time to add one.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**